### PR TITLE
Handle different slopes in mmaps

### LINKED ISSUE #24644: [3.3.5] Blink
**Description:**

First of all, really good job on the updates in https://github.com/TrinityCore/TrinityCore/commit/c0b75bf40da8b447a2f3533dabbfad9eba7c6ea1

Issues where you could blink underground/behind walls appear to be fixed completely, i just noticed that blink appears to have gotten an unusually large buff haha

https://drive.google.com/uc?id=18KQqe867qkUDL6BJbhhDnl1pHmv5zUe8

**Current behaviour:**

You can blink from the bottom of the slope to the top

**Expected behaviour:**

Im not 110% sure if this is a bug, it just feels a bit overpowered :P

**Steps to reproduce the problem:**

1. .go xyz 1398.534 1512.965 325.628 489 (orientation facing towards the slope)
2. cast blink
3. You are at the top of the slope

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

https://github.com/TrinityCore/TrinityCore/commit/8a9f80909387615fb7e22a0f8a042c3ad7a10cfc

**Operating system:** Windows 10

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Handle different slopes in mmaps using different area flags that can be included/excluded when querying mmaps. 
- NAV_AREA_GROUND will be used for ground with slope in the range [0, 55] while NAV_AREA_GROUND_STEEP will be used for ground with slope in the range of (55, 70]. This allows to use the [0, 55] range for Players and idle Creatures and to use the (55, 70] range for Creatures in combat/evading (chasing Players in places usually not reachable)
- NAV_AREA_GROUND_STEEP takes priority over NAV_AREA_GROUND
- Remove --maxAngle parameter from mmaps_generator to keep all parameters hardcoded in MapBuilder::GetMapSpecificConfig() (one is not supposed to try random angles anyway)
- It's required to re-generate mmaps

This is how the tile looks in recastdemo. Light blue is NAV_AREA_GROUND, dark blue is NAV_AREA_GROUND_STEEP:
image

Example path going up the hill using both area types:
image

Same path, this time excluding NAV_AREA_GROUND_STEEP:
image

https://github.com/jackpoz/recastnavigation/tree/fixes/3.3.5-blink-55-degrees is the recastdemo branch that supports this

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #24644

**Tests performed:** (Does it build, tested in-game, etc.)
None

**Known issues and TODO list:** (add/remove lines as needed)

- [x] Implement the change in mmaps_generator
- [x] Implement the change in worldserver
- [x] Cleanup the code to reduce changes in recastnavigation
- [x] raise the steep angle from 70° to 85°

### CODE REVIEW

**Comment by illfated** [comment]
Off-topic/curiosity: Which map are the example images taken from, is it some PvP or raid map? I notice something looking like a Night Elf tower, but I am still struggling too much to recognize it.

**Comment by jackpoz** [comment]
@illfated that's WSG battleground at the location of https://github.com/TrinityCore/TrinityCore/issues/24644

**Comment by jackpoz** [comment]
Current status of this PR: it works like a charm but I would like to change area ids to avoid having NAV_AREA_GROUND and RC_WALKABLE_AREA with the same id 63 (that will required to re-generate mmaps yet again but shouldn't affect anything ingame).

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 1]
```diff
@@ -1015,7 +1016,7 @@ namespace MMAP
         config.maxVertsPerPoly = DT_VERTS_PER_POLYGON;
         config.cs = tileConfig.BASE_UNIT_DIM;
         config.ch = tileConfig.BASE_UNIT_DIM;
-        config.walkableSlopeAngle = 80;
+        config.walkableSlopeAngle = 85;
```
Comment (Faq) [inline src/tools/mmaps_generator/MapBuilder.cpp:1019]: just wondering why changed this.

Comment (jackpoz) [reply src/tools/mmaps_generator/MapBuilder.cpp:1019]: 70° and all filters
image

80° and all filters
image

80° and exclude ledge span filter
image

85° and all filters
image

85° and exclude ledge span filter
image

**Comment by jackpoz** [comment]
PR and test realm updated, mobs will try a more direct route to reach the target when in combat.

**Pull request merged**